rz-cmn-srp: Support preempt-rt image kernel

ThanhNguyenRVC opened the pull request:
Change includes:

- Support preempt-rt image kernel

tiennguyenuh committed:
rz-sbc: add meta layer template

Changes include:
    - Added recipes-kernel
    - Added rzg2l-sbc machine conf

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: move linux bitbake file to the correct location

Changes include:
    - Move linux-yocto_5.10.bb to subfolder named linux in
      recipes-kernel

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: update kernel configuration handling

Changes include:
    - Manually copied the defconfig to the build folder as a workaround
      for the issue where the defconfig was not applied correctly.
    - Used in-tree `KBUILD_DEFCONFIG` defconfig` instead of `KMACHINE`.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanh-doduy-bv committed:
rzg2l-sbc: Add recipes-bsp/trusted-firmware-a

Changes include:
    - Add a recipe to support building Trusted Firmware-A, the module that U-Boot depends on.
    - Remove the "append +=" operator because it is not recommended by Yocto.

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>

khanh-doduy-bv committed:
rzg2l-sbc: Add recipes-bsp/firmware-pack

Changes include:
    - Add a recipe to support building Firmware Pack, the module that U-Boot depends on.

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>

khanh-doduy-bv committed:
rzg2l-sbc: Add recipes-bsp/uboot

Changes include:
    - Add recipes to support build U-boot
    - Remove unused variables in U-Boot recipes.

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>

preetam-reddy committed:
RZPI: Add template layer for 6.10 kernel

Changes include:
	- Clear everything and start clean.
	- Add layer for kernel 6.10 from poky github.
	- Add config features.
	- Add dependencies to get the kernel to build.
	- Create / update readmes's.
	- Backup

Caveats:
	- Experimental. Work in progress. Config is incomplete.

Signed-off-by: Preetam S Reddy <preetam.reddy.cp@renesas.com>

tiennguyenuh committed:
rz-sbc: support for minimal boot

Changes include:
    - Remove `KMACHINE` usage in `linux-yocto` recipe to avoid unexpected
      configuration changes with the in-tree defconfig during the merge.
    - Add basic packages and tools for the RZ/G2L-SBC in
      `core-image-renesas-base.inc`.
    - `core-image-minimal` now requires `core-image-renesas-base.inc` as dependency.
    - Update `LAYERDEPENDS_meta-renesas` to include `openembedded-layer`
      and `meta-python` to support the installation of required tools.

Notes:
    - USB, SPI, I2C, CAN, and GPIO interfaces are functioning correctly.
      Ethernet support is still being developed.
    - The GPIO sysfs interface has been deprecated,
      requiring the use of `libgpiod` for GPIO control as
      `/sys/class/gpio` is no longer available.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: update sii.cfg to enable HDMI support

Enable `CONFIG_DRM_DISPLAY_CONNECTOR` for HDMI to work properly
as a mainline kernel dependency.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add DTS patches and device tree definitions for RZ/G2L-SBC

Changes include:
    - Add patches for building DTB and DTBO files for the RZ/G2L-SBC board.
    - Add cfg files for enable features for RZ/G2L-SBC.
    - Add kernel-image and kernel-devicetree images to yocto build.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: install uEnv.txt and Readme to root filesystem

Changes include:
    - Install uEnv.txt and Readme.md to /boot in the root filesystem
      using the uenv_1.0.bb recipe.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: support wic build and install images to partition 1

Changes include:
    - Install overlays, uEnv, kernel images and device tree blobs into
      partition 1.
    - Add a recipe wic to generate WIC image files.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

nam-vo-khac-rvc committed:
rzg2l-sbc: Add templates configuration for rzg2l-sbc

Changes include:
    - Add templates configuration for rzg2l-sbc

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Support to build core-image-bsp

Changes include:
    - Support core-image-bsp build

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Support to build core-image-weston

Changes include:
    - Support for core-image-weston build
    - Support for building Mesa with Panfrost
    - Add graphic recipe and patches to implement rzg2l-du DRM entry points,
      so mesa can be used with it
    - Add packagegroup for gstreamer plugins
    - Add user-defined weston-init config file

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Fix dependency loop error

Changes include:
    - Fix denpendency loop error when build image by modify index of IMAGE_FSTYPES variable

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

nam-vo-khac-rvc committed:
rzg2l-sbc: Support wifi in yocto styhead release

Changes includes:
    - Add meta-networking and meta-summit-radio into yocto build
    - Include Wi-Fi firmware and summit-supplicant-lwb
      for Laird WPA Supplicant support.
    - Add config to remove commit ID from kernel release version
    - Set PREFERRED_RPROVIDER for wpa-supplicant
    - Add core-image-renesas-sbc and core-image-renesas-quickboot.inc

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rzg2l-sbc: add linux patches and config fragment

Changes include:
    - Support the OV5640 camera and DA7219 audio codec.
    - Support Waveshare touch screen.
    - Enable clock and reset support for the WDT2 peripheral.
    - Fix a soft reset issue with Panfrost by restoring power
      domain properties.
    - Update configuration fragments to enable camera, audio
      and panfrost GPU support.
    - Add a V4L2 initialization script.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

nam-vo-khac-rvc committed:
rzg2l-sbc: Support bluetooth in yocto styhead release

Change include:
    - Add bluetooth tool to support bluetooth.
    - Add config bluetooth.

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

nam-vo-khac-rvc committed:
rzg2l-sbc: update build directory hierarchy

Changes include:
    - Disable generating JSON SPDX for image
    - Update Readme files in each recipe to reflect hierarchy changes
    - Modify bootloader-flasher-linux,sd-creator-linux,uload-bootloader-linux files
      to align with the updated output hierarchy structure
    - Clean up image target that stored in DEPLOY_DIR_IMAGE after build
    - Separate each functionality into its own recipe,
      making it easier to add new features or debug,
      e.g., sd-creator-win, sd-creator-linux, uload-bootloader-win,uenv, etc.
    - Install spdx.json file to host/build if `INHERIT` include `create spdx`
    - Always perform the uenv deploy task, as the Readme.md file is required
      for do_image_wic. The Readme.md will be removed once the image is created.

Signed-off-by: Nam Vo Khac RVC <nam.vo.yb@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Porting kernel modules to support codec

Changes include:
    - Porting kernel modules to support kernel 6.10: mmngr, mmngrbuf, vspm, vspmif

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Add user libraries to support codec

Changes include:
   - Adding user libraries that used by codec: vspm, vspmif, mmngr and
mmngrbuf

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Add gstreamer and gstreamer-omx to image

Changes include:
    - Use Gstreamer from Yocto - version 1.24.6
    - Add Gstreamer-omx from Renesas - version 1.18.5
    - Add libomxil library that needed by gstreamer-omx
    - Add meta-rz-codecs and meta-qt5 to bblayers.conf.sample

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: qt5: support core-image-qt target

Changes include:
    - Add qt5 layer to Yocto build
    - Add qt5 demos and libraries
    - Support SDK build for core images target

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rzg2l-sbc: support eSDK build and resolve marco warnings

Changes include:
    - Add new .bbappend for wpa-supplicant: this update to resolve the
      error when performing eSDK due to the wpa_supplicant.service
      installation conflict between summit-supplicant-lwb and wpa-supplicant.
    - Create a directory in SDKPATH used in the eSDK population task.
    - Fixed warnings related to the usage of :append += macros.
      The += operator has been removed to comply with recommended practices.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add qt6 and using BBFILES_DYNAMIC to handle layer

Changes include:
    - Add qt6 libraries
    - Support SDK and eSDK.
    - Add meta-qt6 to bblayers.conf.sample

Note: eSDK is in progress, it does not include QT library in sysroots
after installing eSDK.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: qt: let user choose which Qt will be supported

Changes include:
    - Default Qt version will be Qt6
    - Add a variable 'QT_SUPPORT6' in bblayers to allow users choose
      which Qt version will be supported.
    - Add logic to conditionally inherit populate_sdk_qt5 or populate_sdk_qt6
      based on the 'QT6_SUPPORT' variable.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: remove gst-omx from repo

Changes include:
    - gst-omx will be removed from this repo because it is already
    on meta-rz-codecs. We just need to modify rz-build-scripts to
    build gst-omx.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Add patch to fix waylandsink issue

Change include:
    - Add kernel patch to support more fourcc video format for CRU to
to fix the issue where waylandsink renders data from CSI (v4l2 device)
too slowly (15 fps instead of the required 30 fps).

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh committed:
rz-sbc: modify renesas-core-image-weston and install kernel modules by default

Changes include:
    - Install kernel module in renesas-base image.
    - Modify renesas core-image-weston based on qt version.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add renesas core images and quickboot images

Changes include:
    - Add Renesas quickboot wayland and cli images.
    - Retain networking stack by using script that allows user to enable
      or disable network-related services.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add python libraries and apt source list

Changes include:
    - Add python libraries and pip.
    - Install default APT sources list using Ubuntu 24.04 repositories.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
rz-sbc : add renesas-ubuntu as a new distro

    - Added ubuntu-tiny.conf to define the base configuration.
    - Updated renesas-ubuntu.bb to include necessary dependencies for
the base package.
    - This commit enhances the build process for Ubuntu in
rz-build-scripts.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

tiennguyenuh committed:
rz-sbc: qt: remove pre-installed Qt demo apps and core Qt images

Changes include:
    - Remove pre-installed Qt demo apps.
    - Remove core-image-qt as it is not necessary.
    - Remove unused Qt framework bbappend.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: use systemctl command to detect services status

Use the systemctl command to detect services status instead of relying on
a file for tracking.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: rename rzpi to rzg2l-sbc across all files and recipes

Rename all instances of `rzpi` to `rzg2l-sbc` within source files and recipes.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
rz-v2h: add ustreamer recipe for MXT

    - Added recipes-app/ustreamer/ustreamer.bb to include ustreamer application.
    - Included ustreamer@.service to set up systemd service for ustreamer.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify recipes-kernel folder

- Add build support for rz-cmn MACHINE
- Modify linux-yocto recipes to support 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify recipes-bsp folder

- Add build support for flash writer, ATF, and U-Boot for 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify remaining folders

- Add build support for rz-cmn MACHINE
- Modify Readme to adapt to new MACHINE

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Update new machanism for support targets

- Update new machanism for support targets for trusted-firmware-a, flash-writer and firmware-pack recipes.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Update uEnv.txt file

- Update uEnv.txt file to use board_id variable from U-boot

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh committed:
rz-sbc: update readme files

Update readme to clarify the purpose of each folder.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add docker support option

Changes include:
	- Add docker support option in `local.conf.sample`
	- Update docker.cfg

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: Update renesas-ubuntu.bb

Changes include:
	- Clean duplicate configuration in `renesas-ubuntu.bb` and fix
	  the issue of creating esdk that needs `environment-setup-append.sh`.
	- Add clean up call in `SDK_POSTPROCESS_COMMAND` after building SDK.
	- Move the renesas-ubuntu recipe into each qt version folder.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
gstreamer-omx: change submodule url

Using a patch to update the submodule URL for gst-omx from
https://gitlab.freedesktop.org/gstreamer/common.git to
https://github.com/GStreamer/common.git

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: correct addtask ordering in bsp recipes

Correct addtask ordering in firmware-package recipes

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

ThanhNguyenRVC committed:
rz-cmn: Add recipes for binmake and platform-gen

- Add binmake-native recipe to build the binmake tool for native use.
- Add platform-gen recipe to generate platform information from JSON.

Signed-off-by: Thanh Nguyen thanh.nguyen.wx@bp.renesas.com

ThanhNguyenRVC committed:
rz-cmn: Deploy srec platform-gen binary to target/images/

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>

tiennguyenuh committed:
rz-sbc: python3-pandas: Add "inherit pkgconfig"

Add inherit pkgconfig to fix this issue.

| Did not find pkg-config by name 'pkg-config'
| Found pkg-config: NO
| Run-time dependency python found: NO (tried pkgconfig, pkgconfig and sysconfig)

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rz-cmn: Update Trusted Firmware A repo URL

- Change repo URL from personal repo to global repo.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: add do_prepare_src task

- Add do_prepare_src task to create a build directory for each board so
we can simplify the SRC_URI variable for do_fetch task.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh committed:
rz-sbc: add guide to switch SSH server and add GPG key

Changes include:
	- Add guide to switch from OpenSSH to Dropbear.
	- Install `trusted.gpg` key by default for APT repository.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rz-cmn: Change include files from rzg2l to rz-cmn

- Change include files from rzg2l to rz-cmn on QT5 and QT6 components

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: update uEnv.txt file

- Update uEnv.txt file to correct booting process on multi boards.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh committed:
rz-cmn: Update README and rename Qt framework include file

Changes include:
	- Update the main README to reflect compatibility with new machine targets
	- Rename Qt include file for clarity and consistency with its Qt-specific usage

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: add universal-scripts task

Changes include:
	- Add universal-scripts task to create a host tools directory
	  support for rz-cmn
	- Remove un-use recipes

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: add audio related-config

Add `CONFIG_SND_ALOOP` to the kernel config fragment

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: add Ethernet patch fix

Add Ethernet patch to prevent eth0 from being disabled before eth1.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: enable APT support for minimal-based images

Add `package-management` and `package_deb` to support APT in
minimal-based images.

Weston-based images worked previously because these features are included by
default.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: retain systemd-resolved and timesyncd in quickboot images

Restore DNS and time sync services needed for network and date accuracy.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: linux recipe and configuration

Changes include:
	- Set the default tune to cortexa55 instead of armv8a for
	  optimization on the RZ/G2L's Cortex-A55 core
	- Add docker.cfg to the Linux Yocto recipe whe

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: Update README and add default BB_PRESSURE_MAX_MEMORY setting

Changes include:
	- Add BB_PRESSURE_MAX_MEMORY with a default value of `100000` in
	  local.conf
	- Add a note to inform users that this value should be adjusted
	  based on their system resources
	- Update README for new cortexa55 tune

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: add tkinter to local.conf

Enable tkinter installation for python3

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: modify v2h support patch and add config fragment

 Changes include:
 	- Use renesas_defconfig instead of defconfig
	- Add config for laird modules
	- Modify the patch to support V2H as it not up
	  to date.
	- Organize patches and configs by board for better clarity

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
rz-cmn-srp: alsa-utils: restore and prefer version 1.2.11 from Scarthgap branch

Due to a regression in alsa-utils 1.2.12 causing silent playback during
audio capture on the RZ/V2H-EVK, we restore version 1.2.11 from the
Scarthgap branch and pin it as the preferred version.

The issue stems from a change in WAV header generation for the S24_LE
format (24-bit audio in a 32-bit container), which breaks compatibility
with certain audio drivers.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: enable xwayland support and optimize kernel config

Changes include:
	- Add 'xwayland=true' to the [core] section of weston.ini to
	  enable automatic Xwayland startup
	- Add optimize kernel config fragment

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: kernel: add support camera ov5645

Changes include:
	- Add kernel patch to support the camera OV5645
	- Deploy corresponding device tree blob to the boot partition

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

ThanhNguyenRVC committed:
rz-cmn-srp: docker: Add essential kernel config options to docker.cfg

The following kernel configuration options have been added to
support IPv6 networking and bridge VLAN filtering in Docker:

  - CONFIG_IP6_NF_FILTER
  - CONFIG_IP6_NF_TARGET_REJECT
  - CONFIG_IP6_NF_MANGLE
  - CONFIG_IP6_NF_NAT
  - CONFIG_IP6_NF_TARGET_MASQUERADE
  - CONFIG_BRIDGE_VLAN_FILTERING

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: recipes-kernel: drop unused config in renesas_defconfig

These kernel options are mainly used for Docker and container networking,
including connection tracking and packet filtering. They are not needed
for non-container use cases and have been removed to simplify the kernel
config

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: recipes-kernel: add patch support and config fragments

Changes include:
	- Applied kernel patches to enable additional features on
	  RZ/V2H-EVK and RZ/G2L-EVK boards.
	- Integrated necessary config fragments into renesas_defconfig
	  for proper hardware and feature enablement.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

ThanhNguyenRVC committed:
rz-cmn-srp: Support common V4L2 initialization script

Change includes:
- Add support for the ov5645 camera sensor
- Support for 1280x960 resolution for the RZG2L-EVK, RZV2L-EVK and RZV2H-EVK
- Support common camera initialization script for all devices

Signed-off-by: Thanh Nguyen thanh.nguyen.wx@bp.renesas.com

khanhduydo-rvc committed:
rz-cmn: Update Flash Writer for RZV2H-EVK

- Update Flash Writer for RZV2H-EVK to support writing file .bin to xSPI
Flash

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

sonnguyenxg committed:
rz-cmn: Add script to config audio

Changes include:
	- Add file audio-init.sh into profile.d folder.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

ThanhNguyenRVC committed:
rz-cmn-srp: Update uEnv.txt and platform_info.json for uboot integration

Change includes:

- Remove hard code uboot env from uEnv.txt
- Update platform_info.json to include uboot information
- Support to build platform binary with revisions

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: recipes-kernel: add kernel patches and config fragment

Changes include:
	- Update kernel patches to remove unnecessary configurations and
	  reorder them to ensure correct patch application sequence.
	- Add kernel patch to support missing features:
	  ADC, POE3, POEG, GPT, I3C and RSCI.
	- Add kernel config fragment for enabling related options

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC committed:
universal-scripts: change to release branch

Signed-off-by: Vu Dang <vu.dang.te@renesas.com>

tiennguyenuh committed:
rz-cmn-srp: add kernel patch to fix DSI and CSI on the RZG2L-SBC

Add a kernel patch to fix DSI initialization and CSI streaming for high
resolution on the RZ/G2L-SBC

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
rz-cmn: recipes-multimedia: move audio-init from base image to BSP image; set DVC In to 10%

Changes include:
    - Added audio-init to BSP image
    - Removed audio-init from base image
    - Set DVC In to 10% in audio-init.sh

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

tiennguyenuh committed:
rz-cmn: recipes-docs: update README

Update README files to reflect the new hierarchy changes.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: recipes-kernel: update OV5645 config fragment

Remove accidentally overridden OV5645 config fragment set to 'n'.

This fragment was unintentionally forcing OV5645 support to be disabled
(ov5645=n in the .cfg file), overriding the intended defconfig setting.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-tools: binmake: change to SST repository

Change URL and branch to Renesas-SST repo

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: restructure directory layout and update README

Reorganize the directory hierarchy to introduce new subfolders
for better modularity and clarity:

- Linux: Contains the Linux Image and Device Tree Blobs (DTBs).
- U-Boot: Contains U-Boot binaries without embedded DTBs, with DTBs
  provided separately.
- ATF: Contains common BL2, BL31, and FCONF DTBs.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
apt: move libapt-private and libapt-pkg to multiarch libdir

To align with the structure of APT 3.1.5 in the Ubuntu repositories,
move `libapt-private.so.0.0.0` and `libapt-pkg.so.6.0.0` into
`${libdir}/aarch64-linux-gnu`

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: update revision numbers and uEnv.txt file

Changes include:
	- Update revision numbers for all supported board.
	- Replace the long elif chain for DTB selection with a
          U-Boot command.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-tools: binmake: update platform info in JSON file

Fix redundant field in the platform JSON.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: update docs and apt sources list

Changes include:
	- Update Readme
	- Switched sources list to old-releases since the Oracular
	  repository has reached EOL

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
flash-writer: add RZ/V2H-EVK to Flash Writer build

Update the recipe to build the Flash Writer for RZ/V2H-EVK instead of
using a prebuilt .mot file.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-bsp: flash-writer: modify build workspace

Update the flash-writer recipe to separate build workspace per DISTRO.
This avoids stale artifacts and build failures when switching between
different distros.)

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-cmn: cleanup initramfs and enforce release_clean

- Mark release_clean as [nostamp] to ensure it runs every build.
- Remove initramfs image

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rz-cmn: Add support for RZV2H-RDK

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

long-trinh-rvc committed:
rz-cmn: add support to IMDT's RZV2H-SBC

Add support to a new board RZV2H-SBC.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

tiennguyenuh committed:
recipes-kernel: add usb serial config fragment

Add usb serial config fragment for CP210x and FTDI.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-docs: update documentation

Changes include:
	- Add updated user manual
	- Add updated quick start guide
	- Add updated RZ evaluation license
	- Remove Disclaimer052.pdf as the Mali GPU is no longer used.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
rz-cmn: Add firmware and kernel driver configurations for Wi-Fi modules

Changes include:

    - Append common linux-firmwares in order to support common USB Wifi dongles
    - Enable Wi-Fi drivers in renesas_defconfig
    - Add documentation to add a new specific firmware for non-common devices

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>
Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>

tiennguyenuh committed:
binmake: update platform_info

Set mmcdev to 1 for rzg2l-evk and rzv2l-evk boards.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

preetam-reddy committed:
RZ-CMN-SRP-3.0: Update docs and license

Changes include:
	- Add reviewed docs and license files.
	- No name change => No change to bb file

Signed-off-by: Preetam S Reddy <preetam.reddy.cp@renesas.com>

tiennguyenuh committed:
recipes-tools: update platform_info

Set the default rootfs MMC device to 1 for rzg2l-evk and rzv2l-evk
boards.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-docs: readme: update README

Update the README sections on firmware flashing and boot-mode switches.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
dynamic-layers: qt6: remove qtxmlpatterns

Remove qtxmlpatterns bbappend file as it is no longer used.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-kernel: linux: add common usb serial and usb can

Enable widely used USB-serial bridges and a broad set of USB-CAN adapters

USB-serial:
	- CH341
	- CDC-ACM

USB-CAN adapters:
	- gs_usb (candleLight/CANtact)
	- PEAK PCAN-USB/USB-Pro
	- 8devices
	- EMS
	- esd electronics
	- ETAS ES58x
	- Fintek F81604
	- Kvaser
	- Microchip MCBA
	- Theobroma UCAN

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
recipes-docs: readme: update README

Update the README sections for IMDT RZV2H-SBC.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

tiennguyenuh committed:
rz-cmn-srp-3.0-rc2: update build hierarchy and add otg flashing note

Update the main README file to include build hierarchy and add an OTG flashing note
to highlight prerequisites and setup

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

ThanhNguyenRVC committed:
rz-cmn: Support multiple board device tree overlays

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>

tiennguyenuh committed:
recipes-kernel: add reviewed patch to linux

Add patch to add I3C_SEL macro to switch between I2C2 and I3C

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

preetam-reddy committed:
rz-cmn-srp-3.x: Updated documents

Updated reviewed documents:
	- User manual
	- Quick start guide
	- License

Signed-off-by: Preetam S Reddy <preetam.reddy.cp@renesas.com>

ThanhNguyenRVC committed:
rz-cmn-srp: Support preempt-rt image kernel

Signed-off-by: Thanh Nguyen thanh.nguyen.wx@bp.renesas.com

tiennguyenuh committed:
recipes-docs: readme: update readme file

Update readme to clarify the flash-writer and board identification
binaries

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -102,7 +109,89 @@ do_deploy:append:rz-cmn(){
+# RT variant source tree
+SRT ?= "${WORKDIR}/git-preempt-rt"
+
+do_compile_preempt_rt() {
```
It seems this function only deploys the RT image (Image-preempt_rt) to `${DEPLOYDIR}/target/images/linux` How do we deploy the matching modules (/lib/modules/6.10.14-**rt14**-yocto-standard) for the RT image? I think this is essential

ThanhNguyenRVC replied:
I have verified the preempt_rt image supporting the installation of default kernel modules at lasted commit.

Boot log: [preempt_rt_kernel_modules.txt](https://github.com/user-attachments/files/23468729/preempt_rt_kernel_modules.txt)

sonnguyenxg committed:
renesas: rz-cmn: Add support board Geniatech rs-g2l100

Changes include:
	- Add platform_info.
	- Update recipes Flash-writer, ATF, u-boot and kernel linux.
	- Update readme

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rz-cmn: create README for meta-renesas

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

preetam-reddy committed:
rz-cmn-srp-3.0: User Manual & QSG update

Add verified QSG and User manual documents.

Signed-off-by: Preetam S Reddy <preetam.reddy.cp@renesas.com>

tiennguyenuh commented on recipes-docs/rz-cmn-readme/files/README.md:
````diff
@@ -1589,6 +1589,77 @@ For more examples and ideas, visit:
  https://docs.docker.com/get-started/
 ```
 
+#### 4.1.9. U-Boot: Select Kernel Image via `image_flavor`
````
Please update the guide for how to select the non-preempt kernel Image.

ThanhNguyenRVC replied:
It is updated

tiennguyenuh replied:
It would be better if the section first briefly introduces the three kinds of images, explains their purpose, and then shows how to use them (you already have usage instructions). Start with a high-level description, then go into the details.

ThanhNguyenRVC replied:
I have updated the readme for the description.

tiennguyenuh commented on include/core-image-renesas-base.inc:
```diff
@@ -195,6 +197,9 @@ EXTRA_IMAGEDEPENDS += " rz-cmn-docs "
 IMAGE_INSTALL:append = " kernel-modules"
 
+IMAGE_INSTALL:append = " linux-yocto-modules-preempt-rt"
+IMAGE_INSTALL:append = " linux-yocto-modules-nonpreempt"
```
What is it? I cannot find any related recipes for this.

ThanhNguyenRVC replied:
It is the old implementation. I have removed it

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -102,7 +109,281 @@ do_deploy:append:rz-cmn(){
+    oe_runmake -C "${SRT}" olddefconfig
+    oe_runmake -C "${SRT}" ${KERNEL_IMAGETYPE}
+    oe_runmake -C "${SRT}" modules
+}
```
For all of these manual steps, I recommend creating a custom .bbclass to encapsulate this logic. You may refer to kernel-yocto.bbclass as an example: https://github.com/yoctoproject/poky/blob/styhead/meta/classes-recipe/kernel-yocto.bbclass. This will make the implementation cleaner, easier to maintain, and reusable across variants.

tiennguyenuh replied:
Please check the alignment as well.

ThanhNguyenRVC replied:
I have support the bbclass for custom kernel images.

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -102,7 +109,281 @@ do_deploy:append:rz-cmn(){
+    oe_runmake -C "${SRT}" modules
+}
+
+EXTRA_OEMAKE += "-j16"
```
It's better to just rely on PARALLEL_MAKE variable.

```
EXTRA_OEMAKE += "${PARALLEL_MAKE}"
```

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on recipes-docs/rz-cmn-readme/files/README.md:
```diff
@@ -1589,6 +1589,80 @@ For more examples and ideas, visit:
+| ---------------- | ------------------ |
+| normal (default) | `Image`            |
+| preempt_rt       | `Image-preempt_rt` |
+| nonpreempt       | `Image-nonpreempt` |
```
You should have the description column for these options, e.g:
- normal (default) should be soft realtime
- preempt_rt is hard realtime
- nonpreempt: non-preemptive Image

ThanhNguyenRVC replied:
It is updated

ThanhNguyenRVC committed:
rz-cmn-srp: Recontruct preempt-rt recipes

Signed-off-by: Thanh Nguyen thanh.nguyen.wx@bp.renesas.com

tiennguyenuh committed:
recipes-kernel: linux: refactor linux recipe

Changes include:
	- Override the SRC_URI to use Renesas-SST/linux-rz git repo.
	- Remove duplicated config fragment
	- Use in-tree defconfig

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
flash-writer: correct board name for rzg2l-evk

Correct board name for the rzg2l-evk.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
